File: MgDev/UnitTest/WebTier/DotNet/TestMapGuideApi/ServerAdmin/ServerAdminOperationExecutor.cs
using OSGeo.MapGuide.Test.Common;
using SqliteDotNet;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OSGeo.MapGuide.Test.Operations
{
    /// <summary>
    /// The base class of all test executors using the MgServerAdmin
    /// </summary>
    /// <typeparam name="T">The test executor subclass type</typeparam>
    public abstract class ServerAdminOperationExecutor<T> : PlatformApiTestExecutor
    {
        protected MgServerAdmin _serverAdmin;

        protected ServerAdminOperationExecutor(MgServerAdmin serverAdmin, string unitTestVm)
            : base(typeof(T).Name.ToUpper(), ApiTypes.Platform, unitTestVm)
        {
            _serverAdmin = serverAdmin;
        }
    }
}

File: MgDev/UnitTest/WebTier/Java/src/org/osgeo/mapguide/test/mapguide/operations/admin/ServerAdminOperationExecutor.java
package org.osgeo.mapguide.test.mapguide.operations.admin;

import org.osgeo.mapguide.*;
import org.osgeo.mapguide.test.*;
import org.osgeo.mapguide.test.common.*;

import com.almworks.sqlite4java.*;

public abstract class ServerAdminOperationExecutor extends PlatformApiTestExecutor
{
    protected MgServerAdmin _serverAdmin;

    protected ServerAdminOperationExecutor(String opName, MgServerAdmin serverAdmin, String unitTestVm)
    {
        super(opName, ApiTypes.Platform, unitTestVm);
        _serverAdmin = serverAdmin;
    }
}